src/tables.ts

```diff
@@ -586,6 +718,56 @@ export class Tables {
     return clefs[clef];
   }
 
+  /** Use the provided key to look up a FontInfo in CommonMetrics. **/
+  static lookupMetricFontInfo(key: string): Required<FontInfo> {
+    return {
+      family: Tables.lookupMetric(`${key}.fontFamily`),
+      size: Tables.lookupMetric(`${key}.fontSize`),
+      weight: Tables.lookupMetric(`${key}.fontWeight`),
+      style: Tables.lookupMetric(`${key}.fontStyle`),
+    };
+  }
+
+  /**
+   * Use the provided key to look up a value in CommonMetrics.
+   * @param key is a string separated by periods (e.g., Stroke.text.fontFamily).
+   *        The following keys will be tried before using the default value:
+   *          - Stroke.text.fontFamily
+   *          - Stroke.fontFamily
+   *          - fontFamily
+   * @param defaultValue is returned if the lookup fails.
+   * @returns the retrieved value (or `defaultValue` if the lookup fails).
+   */
+  // eslint-disable-next-line
+  static lookupMetric(key: string, defaultValue?: any | number): any {
```

Comment:
@rvilarl In the next PR, let's continue thinking about how lookupMetric() could be improved. I was particularly confused about the keyParts manipulation / pop near the end of this method.